ionoscloud/resource_laas_pipeline.go

```diff
@@ -0,0 +1,147 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	validation "github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+	"log"
+)
+
+func resourceLaaSPipeline() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: pipelineCreate,
+		ReadContext:   pipelineRead,
+		UpdateContext: pipelineUpdate,
+		DeleteContext: pipelineDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: pipelineImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"log": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"source": {
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "The source parser to be used",
+							ValidateFunc: validation.StringInSlice([]string{"kubernetes", "docker", "systemd"}, false),
+						},
+						"tag": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The tag is used to distinguish different pipelines. Must be unique amongst the pipeline's array items.",
+						},
+						"protocol": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"http", "tcp"}, false),
+						},
+						"public": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"destinations": {
+							Type:     schema.TypeSet,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func pipelineCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(SdkBundle).LaaSClient
+	pipelineResponse, _, err := client.CreatePipeline(ctx, d)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occured while creating a LaaS pipeline: %w", err))
+	}
+	d.SetId(*pipelineResponse.Id)
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsPipelineAvailable)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occured while waiting for the pipeline with ID: %s to become available: %w", *pipelineResponse.Id, err))
```

Comment:
pipelineResponse might be nil in case of an error, so dereferentiating might crash

Reply:
ah, nevermind